authority/provisioner/oidc_options.go

```diff
@@ -0,0 +1,90 @@
+package provisioner
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"text/template"
+	"time"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+)
+
+type ProviderJSON struct {
+	IssuerURL   string   `json:"issuer,omitempty"`
+	AuthURL     string   `json:"authorization_endpoint,omitempty"`
+	TokenURL    string   `json:"token_endpoint,omitempty"`
+	JWKSURL     string   `json:"jwks_uri,omitempty"`
+	UserInfoURL string   `json:"userinfo_endpoint,omitempty"`
+	Algorithms  []string `json:"id_token_signing_alg_values_supported,omitempty"`
+}
+
+type ConfigJSON struct {
+	ClientID                   string           `json:"client-id,omitempty"`
+	SupportedSigningAlgs       []string         `json:"support-signing-algs,omitempty"`
+	SkipClientIDCheck          bool             `json:"-"`
+	SkipExpiryCheck            bool             `json:"-"`
+	SkipIssuerCheck            bool             `json:"-"`
+	Now                        func() time.Time `json:"-"`
+	InsecureSkipSignatureCheck bool             `json:"-"`
+}
```

Comment:
This looks like configuration options that a user should set; we generally use cammelCase for this.

Reply:
Yes, agree. I've held off on changing these so far because that would break the e2e testing and would need changes on their side and documentation. Will be in a follow-up.

Reply:
Changed this in https://github.com/smallstep/certificates/pull/1671.